### PR TITLE
management sdk: message publisher

### PR DESCRIPTION
**Pull request by realwanpengli**

### CODE REVIEW

File: samples/Management/MessagePublisher/ServerHandler.cs [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+                option.ConnectionString = _connectionString;
+                option.ServiceTransportType = _serviceTransportType;
+            }).Build();
+
+            _hubContext = await serviceManager.CreateHubContextAsync(HubName);
+        }
+
+        public async Task StartAsync()
+        {
+            ShowHelp();
```
Comment (vicancy) [inline samples/Management/MessagePublisher/ServerHandler.cs]: why here?

File: samples/Management/MessagePublisher/ServerHandler.cs [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+            try
+            {
+                while (true)
+                {
+                    var argLine = Console.ReadLine();
```
Comment (vicancy) [inline samples/Management/MessagePublisher/ServerHandler.cs]: move console related stuff into `Program`?

File: samples/Management/MessagePublisher/ServerHandler.cs (hunk 1 above)
Comment (realwanpengli) [reply samples/Management/MessagePublisher/ServerHandler.cs]: After publisher started, shows a guide like this:
```
******************************************
send user <User ID List (Seperated by ',')>
send users <User List>
send group <Group Name>
send groups <Group List (Seperated by ',')>
usergroup add <User ID> <Group Name>
usergroup remove <User ID> <Group Name>
broadcast
**********************************************
```

File: samples/Management/MessagePublisher/Program.cs [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+                .SetBasePath(Directory.GetCurrentDirectory())
+                .AddUserSecrets<Program>()
+                .Build();
+
+            app.Description = "Message publisher using Azure SignalR Service Management SDK.";
+            app.HelpOption("--help");
+
+            app.OnExecute(async() =>
+            {
+                var connectionString = connectionStringOption.Value() ?? configuration["Azure:SignalR:ConnectionString"];
```
Comment (chenkennt) [inline samples/Management/MessagePublisher/Program.cs]: does this support environment variable?

File: samples/Management/MessagePublisher/Program.cs [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+                if (string.IsNullOrEmpty(connectionString))
+                {
+                    MissOptions();
+                    return 0;
```
Comment (chenkennt) [inline samples/Management/MessagePublisher/Program.cs]: return something other than 0

File: samples/Management/MessagePublisher/MessagePublisher.cs [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.SignalR;
+using Microsoft.Azure.SignalR.Management;
+
+namespace Microsoft.Azure.SignalR.Samples.Management
+{
+    public class MessagePublisher
+    {
+        public const string Message = "Hello from server";
```
Comment (chenkennt) [inline samples/Management/MessagePublisher/MessagePublisher.cs]: Make message configurable

File: samples/Management/MessagePublisher/Program.cs [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+
+            var connectionStringOption = app.Option("-c|--connectionstring", "Set connection string.", CommandOptionType.SingleValue, true);
+            var serviceTransportTypeOption = app.Option("-t|--transport", "Set service transport type. Options: <transient>|<persistent>. Default value: transient. Transient: calls REST API for each message. Persistent: Establish a WebSockets connection and send all messages in the connection.", CommandOptionType.SingleValue, true); // todo: description
+            var configuration = new ConfigurationBuilder()
+                .SetBasePath(Directory.GetCurrentDirectory())
+                .AddUserSecrets<Program>()
+                .Build();
+
+            app.Description = "Message publisher using Azure SignalR Service Management SDK.";
+            app.HelpOption("--help");
```
Comment (chenkennt) [inline samples/Management/MessagePublisher/Program.cs]: Duplicate?

File: samples/Management/MessagePublisher/Program.cs (hunk 3 above)
Comment (realwanpengli) [reply samples/Management/MessagePublisher/Program.cs]: no, added

File: samples/Management/MessagePublisher/MessagePublisher.cs (hunk 5 above)
Comment (chenkennt) [inline samples/Management/MessagePublisher/MessagePublisher.cs]: Not used, remove

**Pull request merged**